### PR TITLE
enable node for ruby apps

### PR DESCRIPTION
**Pull request by gajicdev**
This commit adds the functionality for the buildpack.

It enables node on runtime for ruby apps.

Related to https://gitlab.nine.ch/ninech/platform/mono-infra/-/issues/853

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Copyright (c) 2013-Present CloudFoundry.org Foundation, Inc. All Rights Reserved.
```
Comment (thirdeyenick) [inline NOTICE]: Not sure this is correct?

File: README.md [hunk 2]
```diff
@@ -1 +1,3 @@
-# buildpack-rails-enable-node
+# buildpack-ruby-enable-node
+
+This is a buildpack that enables node at runtime for ruby apps.
```
Comment (thirdeyenick) [inline README.md:3]: The idea was to use a BUILD env variable to either activate node at runtime or not. Could we add that here?

File: build_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
+package enable
```
Comment (thirdeyenick) [inline build_test.go]: Is this whole file actually needed? In the end, we just log something in the build part.

File: detect.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+		hasPackageJSON, err := fs.Exists(filepath.Join(context.WorkingDir, "package.json"))
+		if err != nil {
+			return packit.DetectResult{}, fmt.Errorf("failed to check for package.json: %w", err)
+		}
+
+		if hasPackageJSON {
+			logger.Debug.Process("node_modules is wanted for runtime")
+			buildPlan.Requires = []packit.BuildPlanRequirement{nodeRequirement, nodeModulesRequirement}
+			buildPlan.Or = []packit.BuildPlan{{
```
Comment (thirdeyenick) [inline detect.go:50]: Do I understand this correctly? It basically means: if the `nodeModulesRequirement` was not provided by some other buildpack, we are ok with just having the`nodeRequirement`?

File: detect_test.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	"testing"
+
+	"github.com/paketo-buildpacks/packit/v2"
+	"github.com/paketo-buildpacks/packit/v2/scribe"
+	"github.com/sclevine/spec"
+
+	. "github.com/onsi/gomega"
+)
+
+func testDetect(t *testing.T, context spec.G, it spec.S) {
```
Comment (thirdeyenick) [inline detect_test.go:16]: I am not sure, but shouldn't we check for the build plan requirements in this test somehow? So if a `package.json` file exists, we also require node_modules for example.

File: integration/default_test.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+			Expect(err).NotTo(HaveOccurred(), logs.String())
+
+			container, err = docker.Container.Run.Execute(image.ID)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(logs).To(ContainLines(
+				MatchRegexp(`changing permission for directories:`),
+			))
+
+			// TODO: actually check directories
```
Comment (thirdeyenick) [inline integration/default_test.go]: do we need this file? It doesn't really test our stuff, right?

File: integration/testdata/default_app/Gemfile [hunk 7]
```diff
@@ -0,0 +1,3 @@
+source "https://rubygems.org"
```
Comment (thirdeyenick) [inline integration/testdata/default_app/Gemfile]: Do we need this file somewhere?

File: scripts/integration.sh [hunk 8]
```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env bash
```
Comment (thirdeyenick) [inline scripts/integration.sh]: Also not sure about this one here. Currently we don't have any integration tests I'd say.

File: NOTICE (hunk 1 above)
Comment (gajicdev) [reply NOTICE]: took that from here:

https://github.com/ninech/buildpack-rails-permissions/blob/main/NOTICE

File: build_test.go (hunk 3 above)
Comment (gajicdev) [reply build_test.go]: Not really just for logging purposes, but I will remove it so it is cleaner

File: detect.go (hunk 4 above)
Comment (gajicdev) [reply detect.go:50]: I understand it as well like that.

From the Or code:

```
// Or is a list of additional BuildPlans that may be selected by the
// lifecycle
```

File: detect_test.go (hunk 5 above)
Comment (gajicdev) [reply detect_test.go:16]: will check this out

File: NOTICE (hunk 1 above)
Comment (thirdeyenick) [reply NOTICE]: Yeah, not sure if it is correct there either :-)

Comment (gajicdev) [reply NOTICE]: removed it 👍

Comment (gajicdev) [reply NOTICE]: will also remove it there

File: README.md (hunk 2 above)
Comment (gajicdev) [reply README.md:3]: added

File: integration/default_test.go (hunk 6 above)
Comment (gajicdev) [reply integration/default_test.go]: removed

File: scripts/integration.sh (hunk 8 above)
Comment (gajicdev) [reply scripts/integration.sh]: removed

File: integration/testdata/default_app/Gemfile (hunk 7 above)
Comment (gajicdev) [reply integration/testdata/default_app/Gemfile]: removed

File: build_test.go (hunk 3 above)
Comment (gajicdev) [reply build_test.go]: removed

File: detect_test.go (hunk 5 above)
Comment (gajicdev) [reply detect_test.go:16]: modified the test to check if we set the plan and if is correct len when package.json is set

**Review by thirdeyenick — APPROVED** [review]

**Pull request merged**